aps/utils/roxar/grid_model.py

```diff
@@ -510,33 +506,42 @@ def get_zone_layer_numbering(grid):
 
 
 def get_simulation_box_thickness(
-    grid, zone=None, debug_level=Debug.OFF, max_number_of_selected_cells=1000
+    grid,
+    zone_mapping,
+    zone_index_input=None,
+    debug_level=Debug.OFF,
+    max_number_of_selected_cells=1000,
 ):
     # Check if API has simboxthickness access
+    # Input grid3D object, dict with zone number and names,
+    # Return a dict thickness_per_zone
     try:
         thickness_per_zone = {}
         simbox = grid.simbox
         simbox_increments_dict = simbox.cell_increments
         simbox_increments = simbox_increments_dict['z_increments']
-        number_of_layers_per_zone, _, _ = get_zone_layer_numbering(grid)
-        for zindx, nlayer in enumerate(number_of_layers_per_zone):
-            zone_number = zindx + 1
-            thickness_per_zone[zone_number] = simbox_increments[zindx] * nlayer
+        number_of_layers_per_zone = zone_mapping.get_number_of_layers_per_zone()
+        # Find thickness in number of layers per defined zone in the grid
+        # Use zone_code_names to get zone_number for the zone
         if debug_level >= Debug.VERBOSE:
             print(
                 '-- Get simbox thickness using Roxar API for RMS version 14.1.0 or later'
             )
+        for zindx, nlayer in enumerate(number_of_layers_per_zone):
```

Comment:
I would prefer `z_index, n_layer`, but that is a matter of taste.

Reply:
ok fixed